staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -133,6 +133,10 @@ import (
 // state, except that its ResourceVersion is replaced with a
 // ResourceVersion in which the object is actually absent.
 type SharedInformer interface {
+	// SetName can be called before Run to modify the name of the informer.
+	// Returns true on success, false if called after Run.
+	SetName(string) bool
+
```

Comment:
Is the ability to mutate name after construction necessary for your usage case? My intuition would be that the field added to `SharedIndexInformerOptions` was sufficient for naming.

Reply:
just saw this https://github.com/kubernetes/kubernetes/pull/120578#issuecomment-1714591855

Reply:
@MikeSpreitzer can you help me understand your use case a bit more? I wouldve assume the `SharedInformerFactory` would name all of its informers by the name of the resource, possibly with a prefix identifying the factory itself. Is that sufficient for you

Reply:
Do you mean the `SharedInformerOption` at https://github.com/kubernetes/client-go/blob/v0.29.0/informers/factory.go#L54 ? That is in generated code, so changing it involves changing the generator.

That approach would also require work on https://github.com/kubernetes/client-go/blob/v0.29.0/informers/generic.go#L98

The use cases are mentioned at the start of #111790

Reply:
See #122654 for the approach that involves changing the informer generator.